hepcrawl/spiders/desy_spider.py

```diff
@@ -162,23 +142,10 @@ def _filter_xml_files(list_files_paths):
             if xml_file.endswith('.xml')
         )
 
-    def crawl_local_directory(self):
-        file_names = os.listdir(self.source_folder)
-        xml_file_names = self._filter_xml_files(file_names)
-
-        for file_name in xml_file_names:
-            file_path = os.path.join(self.source_folder, file_name)
-            self.logger.info(
-                'Local: Try to crawl local file: {0}'.format(file_path)
-            )
-            yield Request(
-                'file://{0}'.format(file_path),
-                callback=self.parse,
-            )
 
     def crawl_s3_bucket(self):
         input_bucket = self.s3_resource.Bucket(self.s3_input_bucket)
-        existing_files = os.listdir(self.destination_folder)
+        existing_files = [file.key for file in self.s3_resource.Bucket(self.s3_output_bucket).objects.all()]
```

Comment:
this is going to be expensive if the bucket is a bit full, better skip this check, or check it for every file through the API if really needed.

Reply:
Yes it is needed (It was in original - FTP - implementation) to be sure that we are not processing same file twice. I will add check through API then.